src/ebuckets.c

```diff
@@ -0,0 +1,2244 @@
+             * that are "<" EB_BUCKET_KEY(info->now) and not "<=".
+             * But if there is a list behind ebuckets, then this limitation is not
+             * applied and the operator "<=" will be used instead.
+             *
+             * The '-1' in case of list brings makes both cases aligned to have
+             * same result */
+            uint64_t now = EB_BUCKET_EXP_TIME(expireRanges[i]) + (ebIsList(eb) ? -1 : 0);
+
+            TimeRange range = {EB_BUCKET_EXP_TIME(startRange), EB_BUCKET_EXP_TIME(expireRanges[i]) };
+            ExpireInfo info = {
+                    .maxToExpire = 0xFFFFFFFF,
+                    .onExpireItem = toExpireItemCb,
+                    .ctx = &range,
+                    .now = now,
+                    .itemsExpired = 0};
+
+            ebExpire(&eb, &myEbucketsType, &info);
+
+            assert( (eb==NULL && (i + 1 == numRanges)) || (eb!=NULL && (i + 1 < numRanges)) );
+            assert( info.itemsExpired == (uint64_t) ItemsPerRange[i]);
+            startRange = expireRanges[i];
+        }
+        assert(eb == NULL);
+        assert( (expItemsHashValue & 0xFFFFFFFF) == 0);
+    }
+    ebDestroy(&eb, &myEbucketsType, NULL);
+    gettimeofday(&timeAfter, NULL);
+    timersub(&timeAfter, &timeBefore, &timeDestroy);
+
+    if (printStat) {
+        printf("Time elapsed ebuckets creation     : %ld.%06ld\n", (long int)timeCreation.tv_sec, (long int)timeCreation.tv_usec);
+        printf("Time elapsed active-expire dry-run : %ld.%06ld\n", (long int)timeDryRun.tv_sec, (long int)timeDryRun.tv_usec);
+        if (isExpire)
+            printf("Time elapsed active-expire         : %ld.%06ld\n", (long int)timeDestroy.tv_sec, (long int)timeDestroy.tv_usec);
+        else
+            printf("Time elapsed destroy               : %ld.%06ld\n", (long int)timeDestroy.tv_sec, (long int)timeDestroy.tv_usec);
+    }
+
+}
+
+#define UNUSED(x) (void)(x)
+#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
+
+int ebucketsTest(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    srand(0);
+
+    int verbose = (flags & REDIS_TEST_VERBOSE) ? 2 : 1;
+    UNUSED(verbose);
+
+#ifdef EB_TEST_BENCHMARK
+    TEST("ebuckets - benchmark 10 million items: alloc + add + activeExpire") {
+        /* All mapped to same EB_BUCKET_KEY() */
+        //uint64_t expireRanges[] = { 1805092100000, 1805092100001};      // 1 msec distribution
+        //uint64_t expireRanges[] = { 1805092100000, 1805092101000};      // 1 sec distribution
+        //uint64_t expireRanges[] = { 1805092100000, 1805092160000};      // 1 min distribution
+        //uint64_t expireRanges[] = { 1805092100000, 1805095700000};      // 1 hour distribution
+        //uint64_t expireRanges[] = { 1805092100000, 1805178500000};      // 1 day distribution
+        //uint64_t expireRanges[] = { 1805092100000, 1805696900000};      // 1 week distribution
```

Comment:
need to retain?

Reply:
The context of the comment is rather clear. I prefer to leave it approachable to users. 
At least as long as we are not done with performance evaluation phase.

Reply:
i think the way we can add some commet to describe it, instead of annotating it with // which doesn't allow to be used in redis.